### DISCUSSION TITLE
Поле с адресом (автозаполнение или интерактивная карта)?

### DISCUSSION THREAD

**Discussion by LittleBomba** [totumonline/totum-ru-issues-and-discussions#506] [Как это сделать?]
Есть необходимость в  указании адресов в карточках заказчиков. Что бы не допускать ошибок при ручном вводе пользователем - пошел напролом и создал простую таблицу с всеми населенными пунктами РФ (спасибо, что в сети есть такие готовые таблицы). 
Позже дошло, что нужно еще указывать улицы, дома, корпуса и тд..и понял, что будет очень много таблиц и очень большая нагрузка на сервер тотум при работе поиска в селектах.
Как это можно реализовать рационально?
Первая мысль - подключить интерактивную карту того же Яндекса, и через нее производить поиск адресата и уже потом импортировать полный адрес в нужное поле. Возможна такая комбинация? И где черпать информацию по подключению Api яндекс карт к тотуму или можете показать функционирующий пример? 
Так же приветствую советы с другими способами, что бы достичь максимальной оптимизации и минимальной нагрузки на сервер.

**Comment by LostAlly** [comment]
Мне кажется самым верным будет обращаться не к Яндексу, а к государственному сервису.
У них и API есть https://fias.nalog.ru/Frontend а уже хранить у себя или нет и в каком виде это решать вам )
Подключить карту для визуального уточнения поиска было бы конечно супер ) как это делается в яндекс доставке или достависте.
Но я пока практически не разбираюсь в тотуме, чтобы оценить возможность такой реализации.

**Comment by ttm-alexeypolunin** [comment]
Смотреть в сторону dadata.ru там платнй api, но он делает, то что нужно. Открывать надо через linkToInput с их js и кастомным возвратом данных. Они возвращают json в котором разнесены все параметры адреса.

будет время на неделе, я сделаю пример.

**Comment by LittleBomba** [comment]
Спасибо за подсказки, Даже не знал о существовании данных сервисов. Что у ФИАС, что у dadata автозаполнение адресов бесплатно. Пошел изучать, но в любом случае буду благодарен примеру, что бы понять принцип подключения и использования.

**Reply by erpic-ru** [reply]
https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/327

**Reply by LittleBomba** [reply]
Да, спасибо. Только что сам вышел на этот пост.
